src/k8s/pkg/k8sd/app/hooks_start.go

```diff
@@ -0,0 +1,16 @@
+package app
+
+import (
+	"github.com/canonical/k8s/pkg/k8sd/controllers"
+	"github.com/canonical/k8s/pkg/snap"
+	"github.com/canonical/microcluster/state"
+)
+
+func onStart(s *state.State) error {
+	snap := snap.SnapFromContext(s.Context)
+
+	configController := controllers.NewNodeConfigurationController()
+	go configController.Run(s.Context, snap)
+
```

Comment:
i'd love us to have an option like `--disable-node-configuration-controller` in k8sd. Just a note for the future, don't do this just yet